src/flag_gems/ops/cov.py

```diff
@@ -0,0 +1,90 @@
+import logging
+import torch
+import triton
+import triton.language as tl
+import triton.testing
+
+@triton.jit
+def mean_kernel(
+    X, 
+    mean, 
+    M, 
+    N, 
+    weights,
+    BLOCK_SIZE: tl.constexpr
+):
+    row = tl.program_id(0)
+    block_id = tl.program_id(1)
+    cols = block_id * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+    mask = cols < N
+    x = tl.load(X + row * N + cols, mask=mask, other=0.0)
+    w = tl.load(weights + cols, mask=mask, other=1.0) 
+    weighted_x = x * w
+    sum_x = tl.sum(weighted_x)
+    tl.atomic_add(mean + row, sum_x)
+
+@triton.jit
+def covariance_kernel(
+    X, 
+    cov_matrix, 
+    mean, 
+    M, 
+    N, 
+    weights,
+    BLOCK_SIZE: tl.constexpr
+):
+    row = tl.program_id(0)
+    col = tl.program_id(1)
+    block_id = tl.program_id(2)
+    cols = block_id * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+    mask = cols < N
+
+    x_row = tl.load(X + row * N + cols, mask=mask, other=0.0) - tl.load(mean + row)
+    x_col = tl.load(X + col * N + cols, mask=mask, other=0.0) - tl.load(mean + col)
+
+    w = tl.load(weights + cols, mask=mask, other=1.0)
+    cov = tl.sum(w * x_row * x_col)
+    tl.atomic_add(cov_matrix + row * M + col, cov)
+
+def cov(X, correction=1, fweights=None, aweights=None):
+    logging.debug("GEMS ")
+
+    if not X.is_cuda:
```

Comment:
Gems is behind the Pytorch dispatcher and doesn't need to handle non-cuda inputs.